components/common/SmallButton.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import styled from 'styled-components';
+
+interface ButtonProps {
+  text: string;
+  disabled: boolean;
+  onClick?: (e?: React.MouseEvent<HTMLButtonElement>) => void;
+}
+
+export default function SmallButton({ text, disabled, onClick }: ButtonProps) {
+  return (
+    <GlobalStyledButton disabled={disabled} onClick={onClick}>
+      {text}
+    </GlobalStyledButton>
+  );
+}
+
+const GlobalStyledButton = styled.button`
+  width: calc(50% - 0.6rem);
+  padding: 1.5rem 0 1.4rem;
+
+  border: ${(props) => (props.disabled ? '0.1rem solid' : '0.2rem solid')};
+  border-color: ${(props) => (props.disabled ? ({ theme }) => theme.colors.gray3 : ({ theme }) => theme.colors.main)};
+  border-radius: 0.8rem;
+  background-color: ${(props) =>
+    props.disabled ? ({ theme }) => theme.colors.gray0 : ({ theme }) => theme.colors.main_opacity20};
+  color: ${(props) => (props.disabled ? ({ theme }) => theme.colors.white : ({ theme }) => theme.colors.main)};
```

Comment:
여기같은 경우에는 여러 속성이 disabled 라는 props를 받아서 가독성이 떨어지는 측면이 있는데, 가변스타일링을 이렇게 간소화해도 괜찮을 것 같아 !! 
```suggestion
import styled , {css} from 'styled-components'

${{disabled}=> disabled && css `
       border: 0.2rem solid;
       border-color: ${theme.colors.main};
       background-color: ${theme.colors.main_opacity20};
       color: ${theme.colors.main};
`}

${{disabled} => disabled===false && css `
       border: 0.1rem solid;
       border-color: ${theme.colors.gray3};
       background-color: ${theme.colors.gray0};
       color: ${theme.colors.white};
` }
```

대강 요런식으로!!

Reply:
와 이런거 처음봤는데 너무 깔끔하다!!!

Reply:
오 그러네 ~ 수정해서 올릴게!